erts/emulator/nifs/unix/unix_prim_file.c

```diff
@@ -284,6 +286,11 @@ Sint64 efile_readv(efile_data_t *d, SysIOVec *iov, int iovlen) {
 
         if(use_fallback) {
             result = read(u->fd, iov->iov_base, iov->iov_len);
+
+            /* Some OSs (e.g. macOS) does not allow reads greater than 4 GB,
```

Comment:
Here and elsewhere: I think you meant "2 GB", as that is what was observed in the issue and also fits with overflowing a signed 32-bit int.

Reply:
thanks, fixed!